eng/tools/azure-sdk-tools/azpysdk/verify_whl.py

```diff
@@ -0,0 +1,214 @@
+    cleanup(extract_location)
+    unzip_file_to_directory(zip_file, extract_location)
+    return extract_location
+
+
+def verify_whl_root_directory(dist_dir: str, expected_top_level_module: str, parsed_pkg: ParsedSetup, executable: str) -> bool:
+    # Verify metadata compatibility with prior version
+    version: str = parsed_pkg.version
+    metadata: Dict[str, Any] = extract_package_metadata(get_path_to_zip(dist_dir, version))
+    prior_version = get_prior_version(parsed_pkg.name, version)
+    if prior_version:
+        if not verify_prior_version_metadata(parsed_pkg.name, prior_version, metadata, executable):
+            return False
+
+    # This method ensures root directory in whl is the directory indicated by our top level namespace
+    extract_location = extract_whl(dist_dir, version)
+    root_folders = os.listdir(extract_location)
+
+    # check for non 'azure' folder as root folder
+    non_azure_folders = [d for d in root_folders if d != expected_top_level_module and not d.endswith(".dist-info")]
+
+    if non_azure_folders:
+        logging.error(
+            "whl has following incorrect directory at root level [%s]",
+            non_azure_folders,
+        )
+        return False
+    else:
+        return True
+
+
+def cleanup(path):
+    # This function deletes all files and cleanup the directory if it exists
+    if os.path.exists(path):
+        if os.path.isdir(path):
+            shutil.rmtree(path)
+        else:
+            os.remove(path)
+
+
+def should_verify_package(package_name):
+    return package_name not in EXCLUDED_PACKAGES and "nspkg" not in package_name and "-mgmt" not in package_name
+
+
+def get_prior_version(package_name: str, current_version: str) -> Optional[str]:
+    """Get prior stable version if it exists, otherwise get prior preview version, else return None."""
+    try:
+        all_versions = retrieve_versions_from_pypi(package_name)
+        current_ver = Version(current_version)
+        prior_versions = [Version(v) for v in all_versions if Version(v) < current_ver]
+        if not prior_versions:
+            return None
+
+        # Try stable versions first
+        stable_versions = [v for v in prior_versions if not v.is_prerelease]
+        if stable_versions:
+            return str(max(stable_versions))
+
+        # Fall back to preview versions
+        preview_versions = [v for v in prior_versions if v.is_prerelease]
+        return str(max(preview_versions)) if preview_versions else None
+    except Exception:
+        return None
+
+
+def verify_prior_version_metadata(
+    package_name: str, prior_version: str, current_metadata: Dict[str, Any], executable: str, package_type: str = "*.whl"
+) -> bool:
+    """Download prior version and verify metadata compatibility."""
+    pip_cmd = get_pip_command(executable)
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        try:
+            is_binary = "--only-binary=:all:" if package_type == "*.whl" else "--no-binary=:all:"
```

Comment:
this argument to `uv pip` is what I feel like might explode. Not certain though.